linera-service/src/chain_listener.rs

```diff
@@ -123,30 +123,29 @@ where
         let client = {
             let mut map_guard = clients.map_lock().await;
             let context_guard = context.lock().await;
-            map_guard
-                .entry(chain_id)
-                .or_insert_with(|| {
-                    let client = context_guard.make_chain_client(storage.clone(), chain_id);
-                    Arc::new(Mutex::new(client))
-                })
-                .clone()
+            let btree_map::Entry::Vacant(entry) = map_guard.entry(chain_id) else {
+                return Ok(());
```

Comment:
Can you add a comment to explain the early return? (Apparently, we don't want to run two client streams for the same chain?)

Reply:
The second sentence looks like the general explanation and the first sentence is an example.